### DISCUSSION TITLE
What about benchmarks?

### DISCUSSION THREAD

**Discussion by MrOxMasTer** [ai/nanoid#495] [General]
Why does nanoid lose out in terms of the number of operations per second to uuid?
![image](https://github.com/user-attachments/assets/ebdc22a4-11ac-4105-898e-b9bfc46e4217)

Before that, it was the opposite:
![image](https://github.com/user-attachments/assets/415a90c2-c42e-4b1e-89f2-a363bc16346e)

And a question. Have there been any comparisons and benchmarks in postgresql using nanoid and 128-bit UUID. I just don't understand if it makes sense for me to use this library other than short id?

**Comment by ai** [comment]
> Why does nanoid lose out in terms of the number of operations per second to uuid?

Because `uuid` uses built-in UUID generator from Node.js, which is more memory efficient because it was written on C.

> I just don't understand if it makes sense for me to use this library other than short id?

Nano ID has 2 reasons to use it over `uuid`:
1. Smaller JS bundle for browser.
2. Shorter IDs, which could be important if they are visible for users.

**Comment by PerryFinn** [comment]
I have the same question. If it's just about package size and readability, that might not be enough for me to choose `nanoid`. On the contrary, I might lean more towards speed.
